packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -29,6 +29,7 @@ contract AssignmentHook is EssentialContract, IHook {
 
     struct ProverAssignment {
         address feeToken;
+        uint64 chainId;
```

Comment:
it doesn't work if we simply add chainid here without verifying it's the same as the L1 contract's configured chainId.